preprocessing-scripts/crop_image_and_detect_aimpoint.py

```diff
@@ -0,0 +1,177 @@
+    torch.jit.ScriptModule
+        Loaded PyTorch model.
+
+    Raises
+    ------
+    Exception
+        If the model cannot be loaded or the download fails.
+    """
+    logger.info(f"Downloading checkpoint from {url}...")
+    response = requests.get(url, stream=True)
+    if response.status_code == 200:
+        logger.info("Checkpoint downloaded successfully. Loading the model...")
+        with tempfile.NamedTemporaryFile(delete=True) as temp_file:
+            for chunk in response.iter_content(chunk_size=8192):
+                temp_file.write(chunk)
+            temp_file.flush()
+            try:
+                model = torch.jit.load(temp_file.name, map_location="cpu")
+                logger.info("Model loaded successfully.")
+                return model
+            except Exception as e:
+                logger.error(f"Failed to load the model: {e}")
+                raise
+    else:
+        logger.error(
+            f"Failed to download the checkpoint. Status code: {response.status_code}"
+        )
+        response.raise_for_status()
+
+
+def main(args: argparse.Namespace) -> None:
+    """
+    Process calibration images and extract focal spots using UTIS.
+
+    This function downloads and processes heliostat calibration data before loading the UTIS model from a given URL.
+    It then applies UTIS to detect focal spots for specified heliostats and measurement IDs.
+    """
+    # Base directory for all outputs
+    output_dir = Path(args.output_dir)
+
+    # Subdirectories for specific types of outputs
+    downloaded_data_dir = output_dir / "downloaded_data"
+    extracted_focal_spots_dir = output_dir / "extracted_focal_spots"
+
+    # Create STAC client and download heliostat data.
+    client = StacClient(output_dir=downloaded_data_dir)
+
+    # Load the UTIS model directly from the URL.
+    loaded_model = load_model_from_url(mappings.UTIS_MODEL_CHECKPOINT)
+
+    # Load single measurement
+    client.get_single_calibration_item_by_id(
+        heliostat_id=args.heliostat,
+        item_id=args.measurement_id,
+        filtered_calibration_keys=args.filtered_calibration,
+    )
+
+    # Define paths for current measurement.
+    image_path = (
+        downloaded_data_dir
+        / args.heliostat
+        / mappings.SAVE_CALIBRATION
+        / f"{args.measurement_id}_raw.png"
+    )
+    calibration_properties_path = (
+        downloaded_data_dir
+        / args.heliostat
+        / mappings.SAVE_CALIBRATION
+        / f"{args.measurement_id}-calibration-properties.json"
+    )
+
+    # Extract target from calibration properties.
+    with open(calibration_properties_path, "r") as file:
+        calibration_data = json.load(file)
+    target = calibration_data[mappings.TARGET_NAME_KEY]
+
+    # Crop the image using template matching.
+    cropped_image = crop_image_with_template_matching(image_path, target, n_grid=256)
+
+    # Convert to grayscale by using only green color channel and convert to torch tensor.
```

Comment:
Just asking out of curiosity: Why is only the green color channel used in this conversion?